cogs/birthday/birthday.py

```diff
@@ -433,12 +433,49 @@ async def getSelfBirthday(self, ctx: Context):
                 if month and day:
                     birthday = date(2020, month, day)
                     birthdayStr = "{0:%B} {0:%d}".format(birthday)
-                    await ctx.author.send(
-                        f"{headerGood}: Your birthday is "
-                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
-                    )
-                    return
-
+                    try:
+                        await ctx.author.send(
+                            f"{headerGood}: Your birthday is "
+                            f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}."
+                        )
+                        return
+                    except discord.Forbidden:
+
+                        def check(msg: discord.Message):
+                            return msg.author == ctx.author and msg.channel == ctx.channel
+
+                        try:
+                            await ctx.send(
+                                "You have disabled DMs from this server. Would you still like to continue here? "
+                                "The your birthday will be sent here and deleted after a short delay. "
+                                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
```

Comment:
Can declare a `headerWarn` like [this](https://github.com/SFUAnime/Ren/blob/1efd98750d00fb1e1829ded85a48c826604cf284/cogs/birthday/birthday.py#L479) to use here as a prefix. Also providing a bit of context and placing the action prompt at the end is better I believe.
```suggestion
                                f"{headerWarn}: "
                                "I wanted to DM you the birthday, "
                                "but you have DMs disabled for this server."
                                "Would you like me to send it here? "
                                "The message containing your birthday will be deleted "
                                "after a short delay for your privacy. "
                                f"Type {bold(inline('yes'), escape_formatting=False)} to accept, "
                                f"or {bold(inline('no'), escape_formatting=False)} otherwise."
```
(Can just combine some lines. I wrapped those lines because if it's too long `black` would complain but I'm not sure if it's too long for `black` when doing review.)

Reply:
I reworked the message to be a little bit more clear but I'd still like your input for the corrected version.